sage-ios/SAGE/SAGE/CheckinViewController.swift

```diff
@@ -171,6 +166,48 @@ class CheckinViewController: SGViewController {
     
     @objc private func userPressedEndSession() {
         //Verify location
+        if !self.currentLocation.recent() {
+            self.checkinView.presentActivityIndicator()
+            self.startGettingCurrentLocation()
+            semaphore = dispatch_semaphore_create(0)
+            dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER)
```

Comment:
might be better to have a banner that shows up (we have existing implementation for banners) that says "Updating Location..." so the user can still do stuff